integration-test/background/storage.js

```diff
@@ -19,73 +17,111 @@ describe(`On https://${testPageDomain}/privacy-protections/storage-blocking/`, (
             },
             { polling: 100, timeout: 10000 }
         )
+        return { browser, page }
+    }
+
+    describe(`On https://${testPageDomain}/privacy-protections/storage-blocking/`, () => {
+        let cookies = []
 
-        let iframeFullyLoaded = false
-        try {
-            page.on('requestfinished', (req) => {
-                // once we see this url, we can consider the test completed
-                if (req.url().startsWith(`https://${thirdPartyTracker}/set-cookie`)) {
-                    iframeFullyLoaded = true
+        beforeAll(async () => {
+            let iframeFullyLoaded = false
+            const { browser, page } = await setup()
+            try {
+                page.on('requestfinished', (req) => {
+                    // once we see this url, we can consider the test completed
+                    if (req.url().startsWith(`https://${thirdPartyTracker}/set-cookie`)) {
+                        iframeFullyLoaded = true
+                    }
+                })
+                // Load the test pages home first to give some time for the extension background to start
+                // and register the content-script-message handler
+                await page.goto(`https://${testPageDomain}/`, { waitUntil: 'networkidle0' })
+                await page.bringToFront()
+                await page.goto(`https://${testPageDomain}/privacy-protections/storage-blocking/?store`, { waitUntil: 'networkidle0' })
+                await page.bringToFront()
+                // eslint-disable-next-line no-unmodified-loop-condition
+                while (!iframeFullyLoaded) {
+                    await wait.ms(100)
                 }
-            })
-            // Load the test pages home first to give some time for the extension background to start
-            // and register the content-script-message handler
-            await page.goto(`https://${testPageDomain}/`, { waitUntil: 'networkidle0' })
-            await page.goto(`https://${testPageDomain}/privacy-protections/storage-blocking/?store`, { waitUntil: 'networkidle0' })
-            // eslint-disable-next-line no-unmodified-loop-condition
-            while (!iframeFullyLoaded) {
-                await wait.ms(100)
+                // collect all browser cookies
+                do {
+                    await wait.ms(1000) // allow cookies to be set
```

Comment:
If the page itself was reporting somehow that it's "done" setting everything, would it be enough to get rid of those waits? One thing that I think I messed up with privacy test pages is not providing a clear signal from the page itself that the test is "done".

Reply:
In this case I'm not 100% sure - in this case we're waiting until we can read the cookies set from the browser's cookie DB, and I think there would still be a chance of a race if we called `Network.getAllCookies` in the same tick as the page set the cookies - I'm not sure how quickly the values get flushed to the DB.